lib/improver/tests/utilities/test_cube_units.py

```diff
@@ -0,0 +1,469 @@
+            np.ones((5, 5), dtype=np.float32), spatial_grid='equalarea',
+            time=datetime(2017, 11, 10, 4, 30))
+        crd = self.cube_non_integer_intervals.coord('projection_x_coordinate')
+        crd.points = crd.points * 1.0005
+
+    def test_time_coordinate_to_hours_valid(self):
+        """Test that a cube with a validity time on the hour can be converted
+        to integer hours."""
+
+        target_units = "hours since 1970-01-01 00:00:00"
+        coord = 'time'
+        cube = self.cube
+        cube_units.DEFAULT_UNITS[coord]['unit'] = target_units
+        expected = 419524
+
+        self.plugin([cube], [coord])
+
+        self.assertEqual(cube.coord(coord).points[0], expected)
+        self.assertEqual(cube.coord(coord).units, target_units)
+        self.assertIsInstance(cube.coord(coord).points[0], np.int64)
+
+    def test_time_coordinate_to_hours_invalid(self):
+        """Test that a cube with a validity time on the half hour cannot be
+        converted to integer hours."""
+
+        target_units = "hours since 1970-01-01 00:00:00"
+        coord = 'time'
+        cube = self.cube_non_integer_intervals
+        cube_units.DEFAULT_UNITS[coord]['unit'] = target_units
+
+        msg = ('Data type of coordinate "time" could not be'
+               ' enforced without losing significant precision.')
+        with self.assertRaisesRegex(ValueError, msg):
+            self.plugin([cube], [coord])
+
+    def test_time_coordinate_to_invalid_units(self):
+        """Test that a cube with time coordinate cannot be converted to an
+        incompatible units, e.g. metres."""
+
+        target_units = "m"
+        coord = 'time'
+        cube = self.cube
+        cube_units.DEFAULT_UNITS[coord]['unit'] = target_units
+
+        msg = 'time units cannot be converted to "m"'
+        with self.assertRaisesRegex(ValueError, msg):
+            self.plugin([cube], [coord])
+
+    def test_time_coordinate_to_hours_float(self):
+        """Test that a cube with a validity time on the half hour can be
+        converted to float hours."""
+
+        target_units = "hours since 1970-01-01 00:00:00"
+        coord = 'time'
+        cube = self.cube_non_integer_intervals
+        cube_units.DEFAULT_UNITS[coord]['unit'] = target_units
+        cube_units.DEFAULT_UNITS[coord]['dtype'] = np.float64
+        expected = 419524.5
+
+        self.plugin([cube], [coord])
+
+        self.assertEqual(cube.coord(coord).points[0], expected)
+        self.assertEqual(cube.coord(coord).units, target_units)
+        self.assertIsInstance(cube.coord(coord).points[0], np.float64)
+
+    def test_time_coordinate_hours_to_seconds_integer(self):
+        """Test that a cube with a validity time in units of hours can be
+        converted to integer seconds."""
+
+        target_units = "seconds since 1970-01-01 00:00:00"
+        coord = 'time'
+        cube = self.cube.copy()
+        cube.coord('time').convert_units(target_units)
```

Comment:
I think this "start" cube wants to be in units of hours?  You've set this up with time in integer seconds; you convert forecast period to hours but then enforce on the time coordinate.

Reply:
Yep, thanks for that.